noggin/controller/authentication.py

```diff
@@ -65,34 +64,31 @@ def login():
 def otp_sync():
     form = SyncTokenForm()
     if form.validate_on_submit():
-        try:
-            ipa = untouched_ipa_client(app)
-            response = ipa.otptoken_sync(
-                user=form.username.data,
-                password=form.password.data,
-                first_code=form.first_code.data,
-                second_code=form.second_code.data,
-                token=form.token.data,
-            )
-
-            if (
-                response.status_code == 200
-                and "Token sync rejected" not in response.text
-            ):
-                flash('Token successfully synchronized', category='success')
-            else:
-                flash(
-                    'Something went wrong trying to sync OTP token.', category='danger'
+        with handle_form_errors(form):
+            try:
+                ipa = untouched_ipa_client(app)
+                response = ipa.otptoken_sync(
+                    user=form.username.data,
+                    password=form.password.data,
+                    first_code=form.first_code.data,
+                    second_code=form.second_code.data,
+                    token=form.token.data,
                 )
-                return redirect(url_for('root'))
 
-        except requests.RequestException as e:
-            app.logger.error(
-                f'A HTTP post error {e.__class__.__name__} happened while syncing a token for user '
-                f'{form.username}: {e}'
-            )
-            flash('Something went wrong trying to sync OTP token.', category='danger')
-            return redirect(url_for('root'))
+                if response.ok and "Token sync rejected" not in response.text:
```

Comment:
Actually this test could be done in the IPA client class method. Check that the request is OK and that the response text does not contain the "Token sync rejected" string in there, and raise a BadRequest error if it does. This way the controller only has to catch the BadRequest error.